internal/corazawaf/transaction.go

```diff
@@ -1295,37 +1295,38 @@ func (tx *Transaction) LastPhase() types.RulePhase {
 // AuditLog returns an AuditLog struct, used to write audit logs
 func (tx *Transaction) AuditLog() *auditlog.Log {
 	al := &auditlog.Log{}
-	al.Parts = tx.AuditLogParts
+	al.Parts_ = tx.AuditLogParts
 
+	var alTransaction auditlog.Transaction
 	for _, part := range tx.AuditLogParts {
 		switch part {
 		case types.AuditLogPartAuditLogHeader:
 			clientPort, _ := strconv.Atoi(tx.variables.remotePort.Get())
 			hostPort, _ := strconv.Atoi(tx.variables.serverPort.Get())
 			// YYYY/MM/DD HH:mm:ss
 			ts := time.Unix(0, tx.Timestamp).Format("2006/01/02 15:04:05")
-			al.Transaction = auditlog.Transaction{
-				Timestamp:     ts,
-				UnixTimestamp: tx.Timestamp,
-				ID:            tx.id,
-				ClientIP:      tx.variables.remoteAddr.Get(),
-				ClientPort:    clientPort,
-				HostIP:        tx.variables.serverAddr.Get(),
-				HostPort:      hostPort,
-				ServerID:      tx.variables.serverName.Get(), // TODO check
+			alTransaction = auditlog.Transaction{
```

Comment:
This code specifically requires the `AuditLogPartAuditLogHeader` part to be present and be the first one declared. The first condition is right as it is mandatory but the second isn't so I will change that in a next PR to not to grow this one.

Reply:
cc @jptosso @Hayak3